task-service/src/main/java/com/infodation/task_service/services/TaskServiceImpl.java

```diff
@@ -21,26 +20,47 @@
 import org.springframework.http.HttpStatus;
 import org.springframework.scheduling.annotation.Async;
 import org.springframework.web.multipart.MultipartFile;
-
 import java.io.BufferedReader;
 import java.io.InputStreamReader;
 import java.text.SimpleDateFormat;
 import java.util.*;
 
 @Service
 public class TaskServiceImpl implements ITaskService {
-
     private final TaskRepository taskRepository;
     private final ITaskCategoryService taskCategoryService;
     private final ITaskStatusService taskStatusService;
     private  final TaskServiceRepository taskServiceRepository;
+    private final TaskAssignmentRepository taskAssignmentRepository;
+    private final RabbitTemplate rabbitTemplate;
     private static final Logger log = LoggerFactory.getLogger(TaskServiceImpl.class);
 
-    public TaskServiceImpl(TaskRepository taskRepository, ITaskCategoryService taskCategoryService, ITaskStatusService taskStatusService, TaskServiceRepository taskServiceRepository) {
+    public TaskServiceImpl(TaskRepository taskRepository, ITaskCategoryService taskCategoryService, ITaskStatusService taskStatusService, TaskServiceRepository taskServiceRepository, TaskAssignmentRepository taskAssignmentRepository, RabbitTemplate rabbitTemplate) {
         this.taskRepository = taskRepository;
         this.taskCategoryService = taskCategoryService;
         this.taskStatusService = taskStatusService;
         this.taskServiceRepository = taskServiceRepository;
+        this.taskAssignmentRepository = taskAssignmentRepository;
+        this.rabbitTemplate = rabbitTemplate;
+    }
+
+    public void assignTaskToUser(TaskAssignmentDTO taskAssignmentDTO) {
+        try {
+            Task task = taskRepository.findById(taskAssignmentDTO.getTaskId())
+                    .orElseThrow(() -> new IllegalArgumentException("Task not found with ID: " + taskAssignmentDTO.getTaskId()));
+            UserTaskAssignment assignment = new UserTaskAssignment();
+            assignment.setUserId(taskAssignmentDTO.getUserId());
+            assignment.setTask(task);
+            sendNotificationToRabbitMQ(taskAssignmentDTO);
+            taskAssignmentRepository.save(assignment);
```

Comment:
You need to send a notification to the RabbitMQ only when saving task into database successfully.
```suggestion
            taskAssignmentRepository.save(assignment);
            sendNotificationToRabbitMQ(taskAssignmentDTO);
```